playbook/app/pb_kits/playbook/utilities/globalProps.ts

```diff
@@ -441,11 +441,54 @@ const PROP_CATEGORIES: {[key:string]: (props: {[key: string]: any}) => string} =
     return css
   },
 
-  topProps: ({ top }: Top) => top ? `top_${top}` : '',
-  rightProps: ({ right }: Right) => right ? `right_${right}` : '',
-  bottomProps: ({ bottom }: Bottom) => bottom ? `bottom_${bottom}` : '',
-  leftProps: ({ left }: Left) => left ? `left_${left}` : '',
-
+  topProps: ({ top }) => {
+    let css = "";
+    if (typeof top === 'string') {
+      css += `top_${top}`;
+    } else if (typeof top === 'object' && top.inset) {
+      css += `top_${top.value}_inset`;
+    } else if (typeof top === 'object' && !top.inset) {
+      css += `top_${top.value}`;
+    }
+    return css;
+  },
+  
+  rightProps: ({ right }) => {
+    let css = "";
+    if (typeof right === 'string') {
+      css += `right_${right}`;
+    } else if (typeof right === 'object' && right.inset) {
+      css += `right_${right.value}_inset`;
+    } else if (typeof right === 'object' && !right.inset) {
+      css += `right_${right.value}`;
+    }
+    return css;
+  },
+  
+  bottomProps: ({ bottom }) => {
+    let css = "";
+    if (typeof bottom === 'string') {
+      css += `bottom_${bottom}`;
+    } else if (typeof bottom === 'object' && bottom.inset) {
+      css += `bottom_${bottom.value}_inset`;
+    } else if (typeof bottom === 'object' && !bottom.inset) {
+      css += `bottom_${bottom.value}`;
+    }
+    return css;
+  },
+  
+  leftProps: ({ left }) => {
+    let css = "";
+    if (typeof left === 'string') {
+      css += `left_${left}`;
+    } else if (typeof left === 'object' && left.inset) {
+      css += `left_${left.value}_inset`;
+    }else if (typeof left === 'object' && !left.inset) {
+      css += `left_${left.value}`;
+    }
+    return css;
+  },
+  
```

Comment:
I know when we chatted about this, you were thinking about abstracting to a smaller function to get things a little more DRY and have a single point of truth.
 
I think you can feel that freedom in this file. 

Do you want to test this: 
```tsx

const generateCss = (prop, value) => {
  let css = "";
  if (typeof value === 'string') {
    css = `${prop}_${value}`;
  } else if (typeof value === 'object' && value.inset) {
    css = `${prop}_${value.value}_inset`;
  } else if (typeof value === 'object' && !value.inset) {
    css = `${prop}_${value.value}`;
  }
  return css;
};

const cssProps = {
  topProps: ({ top }) => generateCss('top', top),
  rightProps: ({ right }) => generateCss('right', right),
  bottomProps: ({ bottom }) => generateCss('bottom', bottom),
  leftProps: ({ left }) => generateCss('left', left),
};
```

Reply:
Done! Slightly different from your above suggestion but same principles! @jasperfurniss